mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingListener.kt

```diff
@@ -92,4 +99,114 @@ class ArmorStandLockingListener : Listener {
             isCancelled = true
         }
     }
+
+    @EventHandler
+    fun EntityMoveEvent.onEntityMove() {
+        /**
+         * This listener is used to prevent gravity and water from affecting armor stands.
+         * This is done by setting the entity's velocity to 0 and cancelling the event.
+         */
+        if (entityType != EntityType.ARMOR_STAND) return
+
+        val armorStand = entity.toGeary().get<LockArmorStand>() ?: return
+
+        if (!armorStand.lockState) return
+
+        // We only care about ArmorStands that have fishing hooks as passengers
+        if (entity.passengers.none { it.type == EntityType.FISHING_HOOK }) return
+
+        entity.velocity = Vector(0.0, 0.0, 0.0)
+        isCancelled = true
+    }
+
+    @EventHandler
+    fun PlayerFishEvent.onFishArmorstand() {
+        /**
+         * This listener is used to prevent players from fishing armor stands if they are not allowed to.
+         * This is done by simply cancelling the event. A player can fish an armor stand if they are allowed to via /mia lock
+         * or if they have the permission to bypass the lock.
+         */
+        if (state != PlayerFishEvent.State.CAUGHT_ENTITY) return // We only care about catching entities
+
+        if (caught == null) return // We only care about non-null caught entities
+        if (caught!!.type != EntityType.ARMOR_STAND) return // We only care about armor stands
+
+        val armorStandEntity = caught as ArmorStand
+        val armorStand = armorStandEntity.toGeary().get<LockArmorStand>() ?: return
+
+        if (!armorStand.lockState) return
+
+        // The player is allowed to fish if they are the owner of the armor stand or if they have the permission to bypass the lock
+
+        if (!armorStand.isAllowed(player.uniqueId) && !player.hasPermission("mineinabyss.lockarmorstand.bypass")) {
+            player.error("You do not have access to fishing this armor stand!")
+            hook.remove()
+            isCancelled = true
+        }
+    }
+
+    private fun onPistonExtendRetract(block: Block, direction: BlockFace, blocks: List<Block>): Boolean {
+        /**
+         * This function is used to prevent pistons from extending armor stands,
+         * to prevent pistons from pushing blocks into armor stands,
+         * and to prevent pistons from pushing armor stands using honey blocks below them.
+         */
```

Comment:
Personally I would leave comments to clear up anything that might cause confusion.
Majority of the PR here is self-explanatory by reading the functions, so the comments kinda just bloat it up abit

Reply:
To add onto boy's note about comments, Javadoc/KDoc goes at the top of the function, not inside. You can learn more about KDoc [here](https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax), but it's essentially just markdown. The extra explanation is less important for events which don't get called anywhere else, but it's fine here.